File: mvn_java_proxy/src/main/java/mystdeim/vertx_examples/mvn_java_proxy/verticle/AccountVerticle.java
package mystdeim.vertx_examples.mvn_java_proxy.verticle;

import static java.lang.System.out;

import io.vertx.core.AbstractVerticle;
import io.vertx.serviceproxy.ServiceBinder;
import mystdeim.vertx_examples.mvn_java_proxy.service.AccountService;

/**
 * @author Roman Novikov
 */
public class AccountVerticle extends AbstractVerticle {

    @Override
    public void start() {
        AccountService service = AccountService.createService(vertx);
        new ServiceBinder(vertx)
                .setAddress(AccountService.ADDRESS)
                .register(AccountService.class, service);
        out.println("AccountVerticle deployed!");
    }
}

File: gradle_kotlin_proxy/src/main/kotlin/mystdeim/vertx_examples/mvn_kotlin_proxy/verticle/AccountVerticle.kt
package mystdeim.vertx_examples.mvn_kotlin_proxy.verticle

import java.lang.System.out

import io.vertx.core.AbstractVerticle
import io.vertx.serviceproxy.ServiceBinder
import mystdeim.vertx_examples.mvn_kotlin_proxy.service.AccountService
import mystdeim.vertx_examples.mvn_kotlin_proxy.service.AccountServiceFactory

/**
 * @author Roman Novikov
 */
class AccountVerticle : AbstractVerticle() {

    override fun start() {
        val service = AccountServiceFactory.createService(vertx)
        ServiceBinder(vertx)
                .setAddress(AccountServiceFactory.ADDRESS)
                .register(AccountService::class.java, service)
        out.println("AccountVerticle deployed!")
    }
}
